### ISSUE TITLE
Trainer interfaces

### ISSUE THREAD

**Issue by Plyb** [DRAGNLabs/MIRROR-Pipeline#13]
**Acceptance Criteria**:
- The following interfaces exist:
  - MirrorDataset: inherits IterableDataset. `__iter__` is typed to return a token tensor and an attention mask tensor, both of shape (T).
  - MirrorModel: inherits nn.Module.
    - `training_step` takes in a batch of tokens (B, T) and a batch of attention masks (B, T), and returns a loss
    - `configure_optimizers` is typed to return an optimizer
- The above interfaces have a single, placeholder implementation each. These should be *very* basic, just returning some random value of the right shapes. Full implementations will come in later tickets.
- Trainer class is created:
  - `fit` takes in a `MirrorDataset` and a `MirrorModel`, and returns nothing. For now, just `pass`, implementation will come later.
  - is also has a `fabric` property that is a `Fabric` instance
- A trainer is created in the main method. If the `fit` subcommand is used, the `fit` method is called with a placeholder model and dataset.
